app/src/main/java/com/ddoczi/tasky/di/MainModule.kt

```diff
@@ -0,0 +1,19 @@
+package com.ddoczi.tasky.di
+
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object MainModule {
+    @Provides
+    @Singleton
+    fun provideCoroutineDispatcher() : CoroutineDispatcher {
+        return Dispatchers.IO
+    }
```

Comment:
How comes you inject the IO dispatcher by default? Could it make sense to keep that flexible based on the needs of the class you inject it in? Something like this
```
interface DispatcherProvider {
    val main: CoroutineDispatcher
    val io: CoroutineDispatcher
    val default: CoroutineDispatcher
    val unconfined: CoroutineDispatcher
}
```